subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1127,6 +1127,98 @@ static int cmd_wifi_ps_wakeup_mode(const struct shell *sh, size_t argc, char *ar
 	return 0;
 }
 
+static int cmd_wifi_promisc_mode(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct net_if *iface = net_if_get_default();
+	struct wifi_promisc_setup promisc_mode = {0};
+	int ret;
+	long mode;
+
+	if (argc == 1) {
+		promisc_mode.oper = WIFI_MGMT_GET;
+	} else if (argc == 2) {
+		promisc_mode.oper = WIFI_MGMT_SET;
+		if (!parse_number(sh, &mode, argv[1], WIFI_PROMISC_DISABLE,
+				WIFI_PROMISC_STA_AP_MODE)) {
+			return -EINVAL;
+		}
+		promisc_mode.mode =  mode;
+	} else {
+		return -EINVAL;
+	}
+
+	promisc_mode.op = WIFI_PROMISC_MODE;
+	ret = net_mgmt(NET_REQUEST_WIFI_PROMISC_SETUP, iface,
+			&promisc_mode, sizeof(promisc_mode));
+	if (ret) {
+		shell_fprintf(sh, SHELL_ERROR, "Unable to perform %s promisc mode operation: %d\n",
+				promisc_mode.oper == WIFI_MGMT_GET ? "get" : "set", ret);
+		return -ENOEXEC;
+	}
+
+	if (promisc_mode.oper == WIFI_MGMT_GET) {
+		shell_fprintf(sh, SHELL_NORMAL, "Wi-Fi current Promisc mode is %d\n",
+				promisc_mode.mode);
+	} else {
+		shell_fprintf(sh, SHELL_NORMAL, "Wi-Fi Promisc mode set to %d\n",
+				promisc_mode.mode);
+	}
+
+	return 0;
+}
+
+static int cmd_wifi_promisc_filter(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct net_if *iface = net_if_get_default();
+	struct  wifi_promisc_setup promisc_filter = {0};
+	int ret;
+	long filter;
+
+	if (argc == 1) {
+		promisc_filter.oper = WIFI_MGMT_GET;
+	} else if (argc == 2) {
+		promisc_filter.oper = WIFI_MGMT_SET;
+		if (!parse_number(sh, &filter, argv[1], WIFI_PROMISC_ALL,
+				WIFI_PROMISC_CTRL)) {
+			return -EINVAL;
+		}
+		promisc_filter.filter = filter;
+	}
+
+	promisc_filter.op = WIFI_PROMISC_FILTER;
+	ret = net_mgmt(NET_REQUEST_WIFI_PROMISC_SETUP, iface,
+			&promisc_filter, sizeof(promisc_filter));
+
+	if (ret) {
+		shell_fprintf(sh, SHELL_ERROR, "Unable to perform %s promisc filter op:%d\n",
+				promisc_filter.oper == WIFI_MGMT_GET ? "get" : "set", ret);
+		return -ENOEXEC;
+	}
+
+	if (promisc_filter.oper == WIFI_MGMT_GET) {
+		shell_fprintf(sh, SHELL_NORMAL, "Wi-Fi Current Promisc filter is set to %d\n",
+				promisc_filter.filter);
```

Comment:
Same suggestion for `wifi_promiscuous_filter2str()`. Unless we allow to combine filters.

Reply:
we can combine filter settings. Hence, a filter2str would not be feasible.